src/Management/src/Endpoint/Actuators/Health/HealthEndpointHandler.cs

```diff
@@ -85,6 +85,10 @@ public async Task<HealthEndpointResponse> InvokeAsync(HealthEndpointRequest heal
                 response.Groups.Add(group);
             }
         }
+        else
```

Comment:
I suspect the condition was inverted (compared to v3) to get rid of an empty branch block. But it wasn't actually correct, leading to this bug. Now that we have found both branches are needed, I would prefer to re-invert the condition to make it easier to read (and the same as v3).

So it becomes:
```c#
if (showDetails == ShowDetails.Never || (showDetails == ShowDetails.WhenAuthorized && !healthRequest.HasClaim))
{
    response.Details.Clear();
}
else
{
    foreach (string group in endpointOptions.Groups.Select(group => group.Key))
    {
        response.Groups.Add(group);
    }
}
```